fuel_utility.py

```diff
@@ -20,89 +20,119 @@
+        else:
+            model_name = model_info
+            model_version = None
+
+        url = '/'.join([FUEL_URI, model_name])
+        response = requests.get(url)
         model = json.loads(response.text)
+
+        if (model_version):
+            model['version'] = model_version
+
         self.models.append(model)
-    
+
     def getByCategory(self, category):
         self.appendModelSearchResults(1, category)
 
-    def getByModel(self, model_name):
-        self.appendSingleModel(model_name)
-    
+    def getByModel(self, model_info):
+        self.appendSingleModel(model_info)
+
     def createDatabaseFile(self, directory):
         database = ET.Element('database')
         name = ET.SubElement(database, 'name')
-        name.text = "Ignition Fuel Model Database"
+        name.text = 'Ignition Fuel Model Database'
         license = ET.SubElement(database, 'license')
-        license.text = "Creative Commons Attribution 3.0 Unported"
+        license.text = 'Creative Commons Attribution 3.0 Unported'
         models = ET.SubElement(database, 'models')
         for i, model in enumerate(self.models):
             uri = ET.SubElement(models, 'uri')
             uri.text = model['name']
-        database_xml = xml.dom.minidom.parseString(ET.tostring(database)) 
-        database_xml = database_xml.toprettyxml()
-        out_file = open(directory+"/database.config", "w")
-        out_file.write(database_xml)
+        self.write_file('%s/database.config' % directory, database)
+
+    def write_file(self, output_filename, root):
+        f = open(output_filename, 'wb')
+        xml_string = ET.tostring(root,
+                                 pretty_print=True,
+                                 encoding='utf-8',
+                                 xml_declaration=True)
+        f.write(xml_string)
+        f.close()
 
     def downloadAndExtractModels(self, directory):
         for model in self.models:
-            if os.path.isdir(directory+"/"+model['name']):
-                logging.info("Model %s already downloaded.", model['name'])
+            if os.path.isdir('/'.join([directory, model['name']])):
+                logging.info('Model %s already downloaded.', model['name'])
             else:
-                logging.info("Downloading %s", model['name'])
-                url = "https://fuel.ignitionrobotics.org/1.0/OpenRobotics/models/"+model['name']+"/1/"+model['name']+".zip"
-                response = requests.get(url, stream=True)
-                z = zipfile.ZipFile(io.BytesIO(response.content))
-                z.extractall(directory+"/"+model['name'])
+                logging.info('Downloading %s', model['name'])
+                if ('version' in model):
+                    model_version = model['version']
+                else:
+                    model_version = "1"
+                url = '/'.join([FUEL_URI,
+                                model['name'],
+                                model_version,
```

Comment:
I am getting the following in the latest revision,
```
Traceback (most recent call last):
  File "fuel_utility.py", line 135, in <module>
    fuel_util.downloadAndExtractModels(arguments['--destination'])
  File "fuel_utility.py", line 105, in downloadAndExtractModels
    model['name']])
TypeError: sequence item 2: expected str instance, int found
```

Explicitly converting model_version to str seems to resolve this